diff --git a/Sources/Socket+File.swift b/Sources/Socket+File.swift
index 81f0c48..ff4b57f 100644
--- a/Sources/Socket+File.swift
+++ b/Sources/Socket+File.swift
@@ -19,15 +19,19 @@ import Foundation
             }
             var writeCounter = 0
             while writeCounter < readResult {
-                #if os(Linux)
-                    let writeResult = send(target, &buffer + writeCounter, readResult - writeCounter, Int32(MSG_NOSIGNAL))
-                #else
-                    let writeResult = write(target, &buffer + writeCounter, readResult - writeCounter)
-                #endif
+                let writeResult = buffer.withUnsafeBytes { (ptr) -> Int in
+                    let start = ptr.baseAddress! + writeCounter
+                    let len = readResult - writeCounter
+#if os(Linux)
+                    return send(target, start, len, Int32(MSG_NOSIGNAL))
+#else
+                    return write(target, start, len)
+#endif
+                }
                 guard writeResult > 0 else {
                     return Int32(writeResult)
                 }
-                writeCounter = writeCounter + writeResult
+                writeCounter += writeResult
             }
         }
     }
